### PR TITLE
Fix version for upload tarball to release

### PR DESCRIPTION
**Pull request by Shaikh-Ubaid**

### CODE REVIEW

File: ci/upload_tarball_to_release.sh [hunk 1]
```diff
@@ -1,16 +1,17 @@
 #!/usr/bin/env bash
 
+lfortran_version=$(<version)
```
Comment (certik) [inline ci/upload_tarball_to_release.sh:3]: Above this you must put `set -ex`. Otherwise it ignores errors.

File: ci/upload_tarball_to_release.sh [hunk 2]
```diff
@@ -1,16 +1,17 @@
...
-lfortran_version=$(<version)
 
 RELEASE_ID=$(\
 curl -s 'https://api.github.com/repos/lfortran/lfortran/releases/latest' | \
 python -c "import sys, json; print(json.load(sys.stdin)['id'], end='')")
 
 echo "RELEASE_ID=$RELEASE_ID"
 
-curl -L \
+curl --fail -L \
```
Comment (certik) [inline ci/upload_tarball_to_release.sh:14]: What does `--fail` do?

Comment (Shaikh-Ubaid) [reply ci/upload_tarball_to_release.sh:14]: It makes the curl command to exit with code other than `0`, when the server response code is above 400

Comment (Shaikh-Ubaid) [reply ci/upload_tarball_to_release.sh:14]: I think I took it from here https://superuser.com/a/657174. It seems curl by default does not exit with `0` even if the request failed.

**Review by certik — APPROVED** [review]

File: ci/upload_tarball_to_release.sh (hunk 2 above)
Comment (certik) [reply ci/upload_tarball_to_release.sh:16]: Perfect.

**Comment by Shaikh-Ubaid** [comment]
It seems to work! Thank you so much for the review, guidance, and approval!

**Comment by certik** [comment]
@Shaikh-Ubaid perfect, great job! This will be a lot more robust way to handle the source tarball.

**Pull request merged**